sn_client/src/wallet.rs

```diff
@@ -280,14 +298,38 @@ impl Client {
                 "sending spend request to the network: {:?}: {spend_request:#?}",
                 spend_request.unique_pubkey()
             );
-            tasks.push(self.network_store_spend(spend_request.clone(), verify_store));
+
+            let the_task = async move {
+                let cash_note_key = spend_request.unique_pubkey();
+                let result = self
+                    .network_store_spend(spend_request.clone(), verify_store)
+                    .await;
+
+                (cash_note_key, result)
+            };
+            tasks.push(the_task);
         }
 
-        for spend_attempt_result in join_all(tasks).await {
-            spend_attempt_result.map_err(|err| WalletError::CouldNotSendMoney(err.to_string()))?;
+        let mut spent_cash_notes = Vec::new();
+        for (cash_note_key, spend_attempt_result) in join_all(tasks).await {
+            // if this is a record mismatch on spend, we need to clean up and remove the spent CashNote from the wallet
+            if let Err(Error::Network(sn_networking::Error::ReturnedRecordDoesNotMatch(
```

Comment:
Also the clean up happens only if `verify_store == true`. Can we add a comment for that. Might save be useful to know that.